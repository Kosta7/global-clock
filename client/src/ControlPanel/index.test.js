import React from "react";
import { shallow, mount } from "enzyme";
import sinon from "sinon";

import { ControlPanel } from "./index";

it("renders correctly", () => {
  const wrapper = shallow(
    <ControlPanel
      isEditMode={false}
      isLoggedIn={false}
      toggleEditMode={() => {}}
      resetShift={() => {}}
      updateClockList={() => {}}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it("calls resetShift on click", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <ControlPanel
      isEditMode={false}
      isLoggedIn={false}
      toggleEditMode={() => {}}
      resetShift={spy}
      updateClockList={() => {}}
    />
  );

  wrapper
    .find("button")
    .at(0)
    .simulate("click");

  expect(spy.calledOnce).toBe(true);
});

it("calls toggleEditMode on click", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <ControlPanel
      isEditMode={false}
      isLoggedIn={false}
      toggleEditMode={spy}
      resetShift={() => {}}
      updateClockList={() => {}}
    />
  );

  wrapper
    .find("button")
    .at(1)
    .simulate("click");

  expect(spy.calledOnce).toBe(true);
});
