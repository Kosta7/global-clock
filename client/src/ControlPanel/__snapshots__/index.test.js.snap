// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.button
    onClick={[Function]}
  >
    <span
      aria-label="reset"
      role="img"
    >
      🔃
    </span>
  </styled.button>
  <styled.button
    isEditMode={false}
    onClick={[Function]}
  >
    <span
      aria-label="edit"
      role="img"
    >
      ✏️
    </span>
  </styled.button>
  <styled.button
    onClick={[Function]}
  >
    login
  </styled.button>
</styled.div>
`;
